tests/end_to_end/test_helper.cpp

```diff
@@ -645,13 +650,15 @@ _test_handle_helper::~_test_handle_helper()
     }
 }
 
-static void
-_rundown_osfhandles()
+_Requires_lock_not_held_(_fd_to_handle_mutex) static void _rundown_osfhandles()
 {
     std::vector<int> fds_to_close;
+
+    std::unique_lock lock(_fd_to_handle_mutex);
     for (auto [fd, handle] : _fd_to_handle_map) {
         fds_to_close.push_back(fd);
     }
+    lock.unlock();
```

Comment:
nit:  We can scope lines 657 to 660, so that we dont need to explicitly call `lock.unlock()` :
```
{
    std::unique_lock lock(_fd_to_handle_mutex);
    for (auto [fd, handle] : _fd_to_handle_map) {
        fds_to_close.push_back(fd);
    }
}
```

Reply:
Sure, like the suggestion!